oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java

```diff
@@ -50,6 +50,16 @@ public FlatFileStore(BlobStore blobStore, File storeFile, NodeStateEntryReader e
         this.compressionEnabled = compressionEnabled;
     }
 
+    public String getFlatFileStorePath() {
+        if (storeFile.exists() && storeFile.isFile() && storeFile.canRead()) {
+            return storeFile.getAbsolutePath();
+        }
+
+        String msg = String.format("Cannot read store file at [%s]",
+                storeFile.getAbsolutePath());
+        throw new IllegalArgumentException(msg);
```

Comment:
I wonder if we could move the exception handling to the setter... Having it in the getter is weird in my view.

Reply:
done